crates/weaver_common/src/lib.rs

```diff
@@ -64,14 +68,16 @@ impl ConsoleLogger {
         ConsoleLogger {
             logger: Arc::new(Mutex::new(paris::Logger::new())),
             debug_level,
+            mute: Arc::new(Mutex::new(false)),
         }
     }
 }
 
 impl Logger for ConsoleLogger {
     /// Logs an trace message (only with debug enabled).
     fn trace(&self, message: &str) {
-        if self.debug_level > 0 {
+        let mute = *self.mute.lock().expect("Failed to lock mute");
```

Comment:
Why do we need to lock? Just check current value.

Also if you set this early, you may even be able to lazy-read in practice...

Reply:
Replaced the mutex by an atomic in [d14e711](https://github.com/open-telemetry/weaver/pull/187/commits/d14e711c0028d10c21b260cbf287db549b6dc8d4)